File: JAVA/Arrays/PlusOne.java
// LeetCode Problem 66 : Plus One
// Time Complexity: O(n)
// Space Complexity: O(1)
class Solution {
    public int[] plusOne(int[] digits) {
        int n=digits.length;
        for(int i=digits.length-1;i>=0;i--){
            if(digits[i]<9){
                digits[i]++;
                return digits;
            }
            digits[i]=0;
        }
        int[] result=new int[n+1];
        result[0]=1;
        return result;
        
    }
}

File: C++/Arrays/PlusOne.cpp
//Time Complexity=O(n)
//Space Complexity=O(1)
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int n=digits.size();
        for(int i=n-1;i>=0;i--){
            if(digits[i]==9){ //if number is 9 make it zero
                digits[i]=0;
            }
            else{
                digits[i]++; //if not 9 add and  return 
                return digits;
            }
        }
        digits.resize(n+1); // if all are 9 inc size of array
        digits[0]=1; // put 1 in starting eg :999->1000
        return digits;
    }
};